lib/optimizely/config_manager/http_project_config_manager.rb

```diff
@@ -185,18 +185,73 @@ def polling_interval(polling_interval)
       #
       # polling_interval - Time in seconds after which to update datafile.
 
-      # If polling interval is less than minimum allowed interval then set it to default update interval.
+      # If valid set given polling interval, default update interval otherwise.
 
-      if (polling_interval.is_a? Integer) && (polling_interval >= Helpers::Constants::CONFIG_MANAGER['MIN_UPDATE_INTERVAL'])
-        @polling_interval = polling_interval
+      if polling_interval.nil?
+        @logger.log(
+          Logger::DEBUG,
+          "Polling interval is not provided. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}."
+        )
+        @polling_interval = Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']
         return
       end
 
-      @logger.log(
-        Logger::DEBUG,
-        "Invalid update_interval #{polling_interval} provided. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}"
-      )
-      @polling_interval = Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']
+      unless polling_interval.is_a? Integer
+        @logger.log(
+          Logger::ERROR,
+          "Polling interval '#{polling_interval}' has invalid type. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}."
+        )
+        @polling_interval = Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']
+        return
+      end
+
+      unless polling_interval.between?(Helpers::Constants::CONFIG_MANAGER['MIN_SECONDS_LIMIT'], Helpers::Constants::CONFIG_MANAGER['MAX_SECONDS_LIMIT'])
+        @logger.log(
+          Logger::DEBUG,
+          "Polling interval '#{polling_interval}' has invalid range. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}."
```

Comment:
measurement unit would be good for all logs
`Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}. seconds`

Reply:
Yep we should specify the unit